search/engine.go

```diff
@@ -483,6 +515,12 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 	// Seach intents and grammars in parallel to native search.
 	intentsChannel := make(chan []Intent)
 	go func() {
+		defer func() {
```

Comment:
You use this pattern of inserting empty result for recover...
What recover does?

What that comes to fix? some kind of deadlock when errors are happening?

Reply:
Go-Routens protection.
Without this pattern, panics inside go-routens are not handled and the whole program crashes.